tests/http/exceptions/test_http_proxy_auth_failed.py

```diff
@@ -8,43 +8,44 @@
     :copyright: (c) 2013-present by Abhinav Singh and contributors.
     :license: BSD, see LICENSE for more details.
 """
+import pytest
 import selectors
-import unittest
-from unittest import mock
+
+from pytest_mock import MockerFixture
 
 from proxy.common.flag import FlagParser
 from proxy.http.exception.proxy_auth_failed import ProxyAuthenticationFailed
 from proxy.http import HttpProtocolHandler
 from proxy.core.connection import TcpClientConnection
 from proxy.common.utils import build_http_request
 
+from ...test_assertions import Assertions
+
 
-class TestHttpProxyAuthFailed(unittest.TestCase):
+class TestHttpProxyAuthFailed(Assertions):
 
-    @mock.patch('selectors.DefaultSelector')
-    @mock.patch('socket.fromfd')
-    def setUp(
-        self,
-        mock_fromfd: mock.Mock,
-        mock_selector: mock.Mock,
-    ) -> None:
-        self.mock_fromfd = mock_fromfd
-        self.mock_selector = mock_selector
+    @pytest.fixture(autouse=True)   # type: ignore[misc]
```

Comment:
One good thing, we are now on `pytest` for the most part :)

Reply:
It's still far from following pytest's common usage patterns.

Reply:
Do you mean without classes?  Anything wrong about `autouse=True` here within class context.

Reply:
It's just something flexible. Plus yes, w/o classes you could also do autouse if really needed.

Reply:
Tracking here https://github.com/abhinavsingh/proxy.py/issues/775